Modules/_threadmodule.c

```diff
@@ -578,6 +578,17 @@ Do note that if the lock was acquire()d several times in a row by the\n\
 current thread, release() needs to be called as many times for the lock\n\
 to be available for other threads.");
 
+static PyObject *
+rlock_locked(rlockobject *self, PyObject *Py_UNUSED(ignored))
+{
+    return PyBool_FromLong(self->rlock_count);
```

Comment:
This would introduce a potential data race in free-threaded builds. When the lock is locked, `rlock_count` may be concurrently modified by the thread that holds the lock. 

```suggestion
    return PyBool_FromLong(_Py_atomic_load_ullong_relaxed(&self->rlock_count) != 0);
```